### PR TITLE
Add JBrowseLinearGenomeView React component

### LINKED ISSUE #1027: embeddable-LGV state model and theme providers
In `packages/embeddable-linear-genome-view`:

Make some factory functions that make it easy to make a state tree root model, theme provider, and all the other tooling that a JB2 view needs in order to run. If you find yourself duplicating tons of code from elsewhere, try to refactor the core models to reduce duplication.

```jsx
let model = stateTreeFactory({
  assembly:{ /* full assembly config snapshot */ },
  tracks:[ /* array of full track snapshotIn */ ],
  plugins:[ /* optional list of runtime plugins to load */ ],
  location:"chr4:12345..234534",
  defaultSession:{
    /* session snapshot just for an LGV */
    bpPerPx: 1,
    displayedRegions: [ /* list of displayed regions */ ]
   },
  onChange:(mst_patch) => { console.log("something changed in here", mst_patch) }
})
let theme = useDefaultJBrowseTheme()

<ThemeProvider theme={theme}>
  <JBrowseLinearGenomeView  model={model}/>
</Theme>
```

This should not spawn any web workers, it will use the main-thread RPC driver. Should not require any webpack worker support.

Main tasks

- create a pluginManager with a set of core plugins that are necessary to make a linear genome view
- create a stateTreeFactory that is "easy to use"
- pass state tree model to the React component
- fix some small display issues when displaying the LGV in this standalone way

### LINKED ISSUE #1339: Feature label on VCF features decodes to [object Set]
Ref from 

s3 amazonaws com_jbrowse org_code_jb2_beta_master_index html_config=test_data%2Fconfig_demo json session=local-6bIpl3m9y

session shared link :) https://s3.amazonaws.com/jbrowse.org/code/jb2/beta/master/index.html?config=test_data%2Fconfig_demo.json&session=share-3e30d280077e690270345d29be90dddc21620ad478568b6856c84cc7791dc3c0&password=17d1952cde2a2094bf4a2179e16f5f4f

### PR DESCRIPTION
**Pull request by garrettjstevens**
This is a React component of the linear genome view. To test it out, go to the `products/jbrowse-react-linear-view` directory and run `yarn storybook`:

image

It can be used like this:

```tsx
import React from 'react'
import {
  createViewState,
  createJBrowseTheme,
  JBrowseLinearView,
  ThemeProvider,
} from '@jbrowse/react-linear-view'

const theme = createJBrowseTheme()

function View() {
  const state = createViewState({
    assembly,
    tracks,
    defaultSession,
    location: 'ctgA:1105..1221',
    onChange: patch => {
      console.log('patch', patch)
    },
  })
  return (
    <ThemeProvider theme={theme}>
      <JBrowseLinearView viewState={state} />
    </ThemeProvider>
  )
}
```

It follows the same pattern as the embedded linear view described in the BCC2020 course, so much of that course's information is relevant for this component: https://jbrowse.org/jb2/docs/bcc2020_embedding_jbrowse_01_getting_started.

Resolves #1027, fixes #1339

### CODE REVIEW

**Comment by cmdcolin** [comment]
I thought we were going to keep the name linear genome view?

**Comment by cmdcolin** [comment]
Btw very cool...will have to try this out...the ability to specify location ctgA:1-100 like that is great

File: plugins/variants/.babelrc [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "presets": [
+    ["@babel/preset-env", { "loose": false }],
```
Comment (cmdcolin) [inline plugins/variants/.babelrc:3]: can you point more detail about what was happening here?

**Comment by rbuels** [comment]
do the recent display mode changes necessitate updating any of the example configs?

**Comment by garrettjstevens** [comment]
It uses the standard example configs, which have been updated, and everything appears to be working.

**Comment by rbuels** [comment]
cool! any plans to auto-deploy the storybook to jbrowse.org? We need to get this auto-deployed I think, could be a separate piece of work from this PR. Let's try to remember to groom that

**Review by rbuels — APPROVED** [review]

File: products/jbrowse-react-linear-genome-view/.storybook/main.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
+const path = require('path')
+const spawn = require('cross-spawn')
+const root = require('find-yarn-workspace-root')()
+
+const pgks = spawn.sync('yarn', ['--json', 'workspaces', 'info'])
```
Comment (cmdcolin) [inline products/jbrowse-react-linear-genome-view/.storybook/main.js:5]: is there a source of this file e.g. was it copied from somewhere? can it be referenced?

**Comment by garrettjstevens** [comment]
This stopped working after merging master. Proposed fix in #1468. Marking as draft again for now.

File: products/jbrowse-react-linear-genome-view/.storybook/main.js (hunk 2 above)
Comment (garrettjstevens) [reply products/jbrowse-react-linear-genome-view/.storybook/main.js:5]: It was inspired by `products/jbrowse-web/rescripts/yarnWorkspacesRescript.js`, I added a comment to that effect.

**Comment by garrettjstevens** [comment]
> We need to get this auto-deployed I think, could be a separate piece of work from this PR.

Issue for tracking that here: #1469

**Comment by garrettjstevens** [comment]
Added a basic test that is intentionally failing. The test should start to pass if #1468 is merged.

**Pull request merged**